math/mathcore/inc/TMathVectorized.h

```diff
@@ -0,0 +1,350 @@
+// @(#)root/mathcore:$Id$
+// Author: Alejandro García Montoro 06/2017
+
+/*************************************************************************
+ * Copyright (C) 2017, CERN                                              *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifdef R__HAS_VECCORE
+
+#ifndef ROOT_TMathVectorized
+#define ROOT_TMathVectorized
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TMathVectorized                                                      //
+//                                                                      //
+// Encapsulate vectorized version of the most frequently used Math      //
+// functions.                                                           //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#include "Rtypes.h"
+#include "TError.h"
+#include "Math/Math_vectypes.hxx"
+//
+#include <algorithm>
+#include <limits>
+#include <cmath>
+#include <VecCore/VecCore>
+
+using ROOT::Double_v;
+using ROOT::Int_v;
+using ROOT::Float_v;
+
+using namespace vecCore;
```

Comment:
Note: ALL using statements (in the global namespace) are forbidden by the coding conventions (for reason similar to those expressed by @amadio )